jogo-oito/src/test/java/controller/GameControllerTest.java

```diff
@@ -0,0 +1,60 @@
+package controller;
+
+import model.GridInterface;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import view.GameUI;
+
+import java.util.Arrays;
+
+import static org.mockito.Mockito.*;
+
+class GameControllerTest {
+
+    private GameController gameController;
+
+    @Mock
+    private GridInterface grid;
+
+    @Mock
+    private GameUI view;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        gameController = new GameController();
+        gameController.setGrid(grid);
+        gameController.setView(view);
+    }
+
+    @Test
+    void testNotifyMoveGameNotComplete() {
+        when(grid.getGridData()).thenReturn(Arrays.asList(1, 2, 3, 4, 0, 5, 6, 7, 8));
+        when(grid.getGameIsCompleteGridPattern()).thenReturn(Arrays.asList(1, 2, 3, 4, 5, 6, 7, 8, 0));
```

Comment:
Bom dia.

Eu atualizei mais instruções no readme original, peço que verifique caso queira continuar.
Muito boa a tua motivação em resolver o desafio. Vi que documentou e detalhou os commits, isso eh muito bom.
Depois do pull request do Allan (Candidato) aceito, deve-se melhorar o código, no sentido de manter ou melhorar a POO. 
Estou apontando aqui pelos valores mágicos 1,2,3,4... 
Se perceber no teu código também aparecem if primitivos.
Você pode fazer outra branch e tentar novamente se desejar, agora não temos limites de pull request por candidatos. 
Atenciosamente.

Reply:
Bom, acredito que agora que sei o que fazer, entrego em 24 horas hahahahahhaahahahahaahaha

Reply:
Bom dia,
Relaxa, pode ir com calma, vc pode aprender, revisar...
Tenha certeza que não é o tempo e sim a qualidade do código que avaliamos.
Recomendo ler este livro Design Patterns: Elements of Reusable Object-Oriented Software.
Não temos falta de candidatos e sim de despreparo de candidato, então capriche.

Espero ter ajudado.

Reply:
O que importa eh a orientação objetos. Me avisa que ai eu faco uma branch para Ruby, para receber teu PR.

Reply:
Se puder criar a branch agradeço